### PR TITLE
Add example for person search

### PR DESCRIPTION
**Pull request by MagicDevilZhang**
## Why this change?
Add demos for person search examples. 

## What does this PR include?
This example demonstrates how to use VQPy to search the target person from videos across different camera views.
- `models.py`: additionally models for person search.
- `main.py`: scripts for person search implemented by VQPy.
- `demo.ipynb/demo.assets`: notebooks and resources ran in Google Colab for testing.

## User API changes
N/A

## How did I test the PR?

Demo notebooks ran in Google Colab, scripts ran locally.

## New dependencies included
N/A

### CODE REVIEW

File: examples/person_search/demo.ipynb [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "select_cons = {\n",
+    "    'track_id': None,\n",
+    "    'candidate': lambda x: str(x[0]),  # convert IDs to string\n",
```
Comment (yushan111) [inline examples/person_search/demo.ipynb:280]: Nit: it seems the name is different from that in `expected_results`, which is "match"

File: examples/person_search/main.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+if __name__ == '__main__':
+    args = make_parser().parse_args()
+
+    # load pre-trained models for person feature extracting
+    from models import ReIDPredictor
+    feature_predictor = ReIDPredictor(cfg="MSMT17/bagtricks_R50.yml") # https://github.com/JDAI-CV/fast-reid/tree/master/configs
+
+    # extract the feature of query images
+    gallery_features = []
+    for file_name in os.listdir(args.query_dir):
```
Comment (yushan111) [inline examples/person_search/main.py:102]: Nit: the name in args is `query_folder`. It might be better if you can keep them consistent

**Comment by yushan111** [comment]
Many thanks for the contribution! It is a terrific example that helps demonstrate VQPy's expressiveness. And thanks a lot for digging into the details of VQPy as well as the formats of our current examples.  

We are working on the PR review and hopefully the PR can be merged into VQPy soon.

**Comment by yushan111** [comment]
One more thing is that we have updated VQPy's installation, and `download_models.sh` no longer exists in our repo, which may break your demo notebook. Would you mind updating your environment preparation to the [new installation](https://github.com/vqpy/vqpy#step-1-install-vqpy) in your notebook?

File: examples/person_search/models.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+def _build_model(cfg, pretrained=True, verbose=True):
+    from torch.hub import (
+        _get_cache_or_reload,
+        load_state_dict_from_url
+    )
+
+    import importlib.util
+
+    repo_dir = _get_cache_or_reload(
```
Comment (hanryxu) [inline examples/person_search/models.py:65]: Two positional args, `trust_repo, calling_fn`, might be required. Looks like the additional arguments are introduced in Pytorch 1.12.

Maybe just call `load_state_dict_from_url` with `model_dir=None` and let Pytorch choose where to cache the model files?

File: examples/person_search/demo.ipynb (hunk 1 above)
Comment (MagicDevilZhang) [reply examples/person_search/demo.ipynb:259]: Thanks! I have checked and updated the expected results. The obsolete key `match` is changed to `candidate`.

File: examples/person_search/models.py (hunk 3 above)
Comment (MagicDevilZhang) [reply examples/person_search/models.py]: Thanks! I have updated the `models.py` from your suggestions:

1. To support the PyTorch 1.12, the `trust_repo` is set to be `check`, which will also be the default value after 1.14. 

2. The `model_dir` is set to None to enable PyTorch to decide the saving path.

**Review by yushan111 — APPROVED** [review]
LGTM. Thanks again for the contribution!

**Review by hanryxu — APPROVED** [review]

**Pull request merged**